db/write_thread.cc

```diff
@@ -781,13 +782,33 @@ void RequestQueue::CommitSequenceAwait(CommitRequest* req,
                                        std::atomic<uint64_t>* commit_sequence) {
   std::unique_lock<std::mutex> guard(commit_mu_);
   while (!requests_.empty() && requests_.front() != req && !req->committed) {
+    // When the subsequent commit finds that the front writer has not yet
+    // submitted, it will help the front writer to perform some tasks
+    auto front = requests_.front()->writer;
+    if (front->multi_batch_writer.pending_wb_cnt.load(std::memory_order_acquire) > 1) {
+      auto claimed = front->multi_batch_writer.claimed_cnt.fetch_add(1, std::memory_order_acquire);
+      if (claimed < front->multi_batch_writer.batches.size()) {
+        guard.unlock();
+        front->multi_batch_writer.ConsumeOne(front, claimed);
+        guard.lock();
+        continue;
+      }
+    } else {
+      // The front writer may be waiting for this helper writer
+      commit_cv_.notify_all();
+    }
```

Comment:
If there a reason why this doesn't work?
```suggestion
    guard.unlock();
    bool res = front->multi_batch_writer.ConsumeOne(front);
    guard.lock();
    if (!res) {
      commit_cv_.notify_all();
    } else {
      continue;
    }
```

Reply:
Yes, I think if we `unlock` before we make sure we already got a valid `claimed`, it may be danger, eg:

There're two writers in `requests queue`,  the front writer thread (short for A),  the other writer (short for B)

T0: `A` has finished its all wbs will enter `CommitSequenceAwait` , `B` is as same as `A`.
T1: B gets lock, A wait for lock.
T2: B gets A's writer (A's stack value), `unlock` but it's interrupted by some reason, eg: IRQ or OS period resched;
T3: A gets lock, pop itself from `requests queue` , and `PebbleWriteImpl` return to its caller (so `front` has been reclaimed);
T4: B call ` front->multi_batch_writer.ConsumeOne` , at this point the front variable is already an invalid value;

Reply:
Okay. I suggest move the atomic operations inside a function, e.g. Claim, Consume.

Reply:
> Okay. I suggest move the atomic operations inside a function, e.g. Claim, Consume.

Done, thanks!